torch/nn/utils/prune.py

```diff
@@ -1100,35 +1103,9 @@ def custom_from_mask(module, name, mask):
     return module
 
 
-def remove(module, name):
-    r"""Removes the pruning reparameterization from a module and the
-    pruning method from the forward hook. The pruned
-    parameter named ``name`` remains permanently pruned, and the parameter
-    named ``name+'_orig'`` is removed from the parameter list. Similarly,
-    the buffer named ``name+'_mask'`` is removed from the buffers.
-
-    Note: 
-        Pruning itself is NOT undone or reversed!
-
-    Args:
-        module (nn.Module): module containing the tensor to prune
-        name (str): parameter name within ``module`` on which pruning
-            will act.
-
-    Examples:
-        >>> m = random_pruning(nn.Linear(5, 7), name='weight', amount=0.2)
-        >>> m = remove_pruning(m, name='weight')
-    """
-    for k, hook in module._forward_pre_hooks.items():
-        if isinstance(hook, BasePruningMethod) and hook._tensor_name == name:
-            hook.remove(module)
-            del module._forward_pre_hooks[k]
-            return module
-
-    raise ValueError(
-        "Parameter '{}' of module {} has to be pruned "
-        "before pruning can be removed".format(name, module)
-    )
+# Lezcano Should we handle this like this?
+def remove(module, name, leave_pruned=True):
```

Comment:
Happy to separate this one into two functions, I think that it can make it more clear from a user interface perspective.

I completely agree that, given the specific requirements of pruning, the `undo` method is tricky to implement on it. On the other hand, I think that the `undo` method can be quite useful in general. I can think of at least one general use case for this in the context of optimization on manifolds, where setting and undoing parametrizations has a very clear geometric meaning.